qiskit/transpiler/passes/optimization/optimize_1q_decomposition.py

```diff
@@ -76,7 +98,7 @@ def run(self, dag):
                 new_circs.append(decomposer(operator))
             if new_circs:
                 new_circ = min(new_circs, key=len)
-                if len(run) > len(new_circ):
+                if len(run) > len(new_circ) or single_u3:
```

Comment:
If the decomposer does not find a `'u1'` or `'u2'` implementation, can we skip the conversion and substitution below? (If a circuit has many U3 gates, I could foresee this being a substantial runtime cost.) Maybe with a check like `new_circ.data[0][0].name != 'u3'`.

Reply:
I see. So you're saying if the decomposer just returns a 'u3', we shouldn't go through this section. Should be
`if len(run) > len(new_circ) or (single_u3 and new_circ.data[0][0].name != 'u3')`. Correct?
And I'll add a comment over line 84 above and rerun the tests to figure out which ones failed without that check.

Reply:
Done in https://github.com/Qiskit/qiskit-terra/pull/5554/commits/4010838784d4234084e5329a37b2e1551ca83b83

Reply:
@kdk. Just found the `ParameterExpression `bug. It's has to do with `circuit.instruction.is_parameterized`. 
```
    def is_parameterized(self):
        """Return True .IFF. instruction is parameterized else False"""
        return any(isinstance(param, ParameterExpression)
                   and param.parameters
                   for param in self.params)
```
The `filter_fn `function in `collect_1q_runs `uses `not is_parameterized `which is only False if at least one of the parameters is unbound. Otherwise `param.parameters` is `set()`. If all are bound, the type remains `ParameterExpression`, but `filter_fn `lets it through. This causes the `np.isclose` in `optimize_1q_decomposition `to fail. The obvious fix is to remove the `ParameterExpression `check there and cast the `params[0] `sent to `np.isclose` to a float.

I'll go ahead and do that and push another commit. I think that will cover everything.